### ISSUE TITLE
note taking support by adding labels/tags 

### ISSUE THREAD

**Issue by wibrt** [asciidocfx/AsciidocFX#493]
feature request

add a extra view on left side, containing tags/lables besides 
![image](https://user-images.githubusercontent.com/17875724/104012016-632dba80-51af-11eb-8d2e-3ae6013f4dfb.png)

partly inspired by https://joearms.github.io/#2019-01-08%20My%20Eureka%20Moment%20with%20the%20TiddlyWiki

the mean goals are:
* to be able to tag/label files
* easily filter and aggregate files based on these tags/labels

hence adding the most important feature of note taking to asciidoxfx

ps: asciidoc markup has also tag and label support, i'm talking from a different perspective here

**Comment by ghorwin** [comment]
I'd also like this feature. Implementation is rather straight-forward, and can be annotated helpfully, if we define the following rules for label usage:

- Label must be placed on top of caption or table/diagram/example title
- no comments in label line
- no comments between label line and following caption

Example:

```markdown
[[sec:requestMasterProtocol]]
#### Aushandel-Protokoll und Vorrangregeln

...

[[ex:myExample]]
.Example for stuff
====
...
====


...

[[fig:myDiagram]]
.Some Diagram
image:SchemaViewTabularView.png[alt="Toggle button tabular view", pdfwith=3cm, left]

```

The text can be parsed for rows starting with [[ and ending with ]] and then the next non-empty line starting either with . or with # chars. The label is extracted and stored together with
filename and line number.

This then gives a list like:

```
List of labels:
  sec:calibrationView            Calibration.adoc:3             '# Calibration View'
  sec:configView                 Config.adoc:3                  '# System Configuration'
  sec:dataGraphView              DataView.adoc:22               '## Data view - Diagram'
  sec:dataTableView              DataView.adoc:61               '## Data view - Table'
  sec:dataView                   DataView.adoc:3                '# Data, Messages and Error Status View'
  sec:errorDiagnostics           ErrorDiagnostics.adoc:2        '# Error Diagnostics and Remote Assistance'
  sec:errorList                  DataView.adoc:96               '### Error List'
  sec:errorView                  DataView.adoc:74               '## System Messages and Error Status View'
  sec:experimentManualMode       Experiment.adoc:80             '## Manual Mode'
  sec:experimentParameterChange  Experiment.adoc:57             '## Modifying Parameters during Experiments'
  sec:experimentView             Experiment.adoc:3              '# Experiments'
  sec:mainMenu                   UserInterface.adoc:56          '## Main Menu'
  sec:parameterView              Parameters.adoc:3              '# Experiment Control Parameters'
  sec:saveData                   DataView.adoc:139              '## Data Export'
  sec:schematicsStatusView       UserInterface.adoc:9           '### Schematic Status View'
  sec:tabularStatusView          UserInterface.adoc:41          '### Tabular Status View'
```

(This output is generated by my AsciiDoc helper script, see https://github.com/ghorwin/asciidoctor-publishing ).

Having this feature natively in AsciiDocFX would be great!

**Comment by rahmanusta** [comment]
It is okay but how can UI look and what features can be helpful?

**Comment by ghorwin** [comment]
@Rahman:

I'd suggest adding a new tab below "Outline" labeled "Labels" or "References".
This tab would show a tree view. 
Top level nodes shows the adoc-file, that references were collected, from. 
Child nodes show: "line xxx, <label>, Caption/title", for example:

```
- intro.adoc
  |
  +-  15, sec:introduction, # Overview and Outline
  +-  22, img:screenshotMainView, Initial view of software
  +- 122, sec:ui_structure, ## User Interface

- fileformats.adoc
  |
  +- 10, sec:fileFormats, # File Formats
  +- 110, subsec:imgFile, ## IMG-files

```

Something like that.

Double-clicking on a reference inserts `<<sec:introduction>>` into text, i.e. a cross reference to the label.

**Comment by ghorwin** [comment]
Yes, indeed, the inverse lookup would be nice, too. This could be a combined tab: "Labels/References", in which you can switch between label treeview as described above, and reference view. Or, simpler to implement, two tabs "Labels" (as described before) and "References".

The latter could show the occurrances of all <<xxxx>> in the text as list or tree view (top node adoc file again, similar to labels view), and highlight those in red, that do not have a matching [[xxxx]] anywhere.

Double-clicking on a reference-node will go to the respective line in the file (and open it if needed).

The "aggregate command" sound interesting, yet probably needs to be specified in more detail.

**Comment by rahmanusta** [comment]
In progress...

<img width="416" alt="image" src="https://github.com/asciidocfx/AsciidocFX/assets/1158804/d35b7cb4-07ad-434c-8c24-6cd988703825">

**Comment by wibrt** [comment]
nice

**Comment by ghorwin** [comment]
looking forward to test it!
